app/models/submission.rb

```diff
@@ -405,6 +415,13 @@ def proquest_agreement=(input)
     self[:proquest_agreement_at] = DateTime.now
   end
 
+  def create_extension_token
+    new_token = SecureRandom.hex(10)
+    new_token = SecureRandom.hex(10) while Submission.exists?(extension_token: new_token)
```

Comment:
I'm curious why this `while` logic is here.  Can't we just store the token in a variable like above?

Reply:
`SecureRandom.hex(10)` generates a random 20 character alphanumeric string, but does not ensure uniqueness. The while checks if there is a submission with that as the token already and generates a new one until it comes up with one not being used. It's a pretty small chance given the number of possibility, but not nonzero, so the while safeguards.